packages/protocol/contracts/tokenvault/BridgedERC721.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.20;
+
+import { console2 } from "forge-std/console2.sol";
+import { Test } from "forge-std/Test.sol";
+import { EssentialContract } from "../common/EssentialContract.sol";
+import { ERC721Upgradeable } from
+    "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+
+contract BridgedERC721 is EssentialContract, ERC721Upgradeable {
+    address public srcToken;
+    uint256 public srcChainId;
+    uint256[47] private __gap;
+
+    error BRIDGED_TOKEN_CANNOT_RECEIVE();
+    error BRIDGED_TOKEN_INVALID_PARAMS();
+
+    /// @dev Initializer to be called after being deployed behind a proxy.
+    // Intention is for a different BridgedERC721 Contract to be deployed
+    // per unique _srcToken i.e. one for Loopheads, one for CryptoPunks, etc.
+    function init(
+        address _addressManager,
+        address _srcToken,
+        uint256 _srcChainId,
+        string memory _symbol,
+        string memory _name
```

Comment:
I really think we need the URI, ignoring the URI goes completely against "Ethereum Equivilance" and anyone who deploys an NFT marketplace/etc on Taiko will have an extra hurdle to support Bridged vs Native tokens. Not sure what the intent here is @dantaik @dantaik I'm sure this conversation has already been hashed out but to me it seems like ignoring the decimals in ERC20.

Reply:
I have two reason not to support it:
- it's more expensive. Usually in an ERC721 token contract, the per-token URL is `baseURL + tokenId.toString()`, but the baseURI is not exposed through an interface function, therefore, we have to query each tokenID's URL and write them in the Bridged contract, so we need one URL write per tokenID.

- the URI may change and it will create data discrepancy.  This will probably more problematic.

Reply:
Lets do it when the UI is absolutely need the URIs on the dest chain.